src/highdicom/pr/content.py

```diff
@@ -737,3 +757,1295 @@ def __init__(
+        if ref_im.SamplesPerPixel == 1:
+            if palette_color_lut is None:
+                raise ValueError(
+                    'For advanced blending presentation, if referenced images '
+                    'are grayscale a palette color lookup table must be '
+                    'provided to pseudo-color the image prior to blending.'
+                )
+        for im in referenced_images:
+            if im.SamplesPerPixel != ref_im.SamplesPerPixel:
+                raise ValueError(
+                    'For advanced blending presentation, all referenced '
+                    'images of an advanced blending item must have the same '
+                    'number of samples per pixel.'
+                )
+            if im.StudyInstanceUID != ref_im.StudyInstanceUID:
+                raise ValueError(
+                    'For advanced blending presentation, all referenced '
+                    'images of an advanced blending item must be part of the '
+                    'same study.'
+                )
+            if im.SeriesInstanceUID != ref_im.SeriesInstanceUID:
+                raise ValueError(
+                    'For advanced blending presentation, all referenced '
+                    'images of an advanced blending item must be part of the '
+                    'same series.'
+                )
+
+        self.BlendingInputNumber = blending_input_number
+
+        ref_im = referenced_images[0]
+        ref_series_uid = ref_im.SeriesInstanceUID
+        ref_im_seq = []
+        for im in referenced_images:
+            series_uid = im.SeriesInstanceUID
+            if series_uid != ref_series_uid:
+                raise ValueError(
+                    'All referenced images must belong to the same series.'
+                )
+            if im.Rows != ref_im.Rows or im.Columns != ref_im.Columns:
+                raise ValueError(
+                    'All referenced images must have the same dimensions.'
+                )
+            if is_tiled_image(ref_im):
+                if (
+                    im.TotalPixelMatrixRows != ref_im.TotalPixelMatrixRows or
+                    im.TotalPixelMatrixColumns != ref_im.TotalPixelMatrixColumns
+                ):
+                    raise ValueError(
+                        'All referenced images must have the same total pixel '
+                        'matrix dimensions.'
+                    )
+            ref_im_item = Dataset()
+            ref_im_item.ReferencedSOPClassUID = im.SOPClassUID
+            ref_im_item.ReferencedSOPInstanceUID = im.SOPInstanceUID
+            ref_im_seq.append(ref_im_item)
+        self.ReferencedImageSequence = ref_im_seq
+        self.StudyInstanceUID = ref_im.StudyInstanceUID
+        self.SeriesInstanceUID = UID()
+
+        # Modality LUT
+        if modality_lut is not None:
+            _add_modality_lut_attributes(
+                self,
+                modality_lut=modality_lut
+            )
```

Comment:
Check that rescale parameters were not provided in this case (I think they should be mutually exclusive, right?)

Reply:
Note that this applies also to the constructors of GSPS and PseudoColorPS

Reply:
I think this is related to the issues with the current implementation of `ModalityLUT` versus `VOILUT`. The former implements an item of the Modality LUT Sequence (which is a `LUT` + one attribute), while the latter implements the VOI LUT macro, which includes the VOI LUT Sequence (where each item is just a `LUT`), **plus** additional attributes such as Window Center and Window Width.

Looking at this again, I think we should do the following:
- Implement `ModalityLUT` and `VOILUT` such that both represent items of the corresponding **sequence items** (for `VOILUT` that would mean just subclassing `LUT` without any additional parameters/attributes)
- Implement `ModalityTransformation` and `VOITransformation` which implement the **macros**, which includes additional attributes. In case of Modality, the macro includes Rescale Intercept and Rescale Slope and in case of VOI it includes Window Center and Window Width.

That would finally make sense in my opinion.

Reply:
In my opinion there are no issues with the current implementation, it is the logical class hierarchy given how things are laid out in the standard, optimizes code re-use and maintainability, and all terminology matches the standard precisely. However the terminology in the standard is highly confusing. The only question is whether we can eliminate some confusion by breaking with the standard and renaming some of the classes because we think that it will be easier for users to understand that way. I'm not completely opposed to this, but I am concerned that by breaking with the terminology in the standard we might cause even confusion for those who are already familiar with it. Note for example that `pydicom` has functions to `apply_modality_lut` and `apply_voi_lut` even though these operations actually apply the relevant *transformation* regardless of whether it is a LUT or not. So users of pydicom may well already be used to the confusing terminology.

Most of the confusion here is due to the naming of what is currently the `VOILUT` class. I definitely think that having this class as implemented makes sense (but maybe under different name). It has two purposes:
- It represents an item of the "FrameVOILUTSequence" in the per-frame functional groups in things like parametric map. This actually represents a transformation, not a LUT.
- It serves as a base class for the `SoftcopyVOILUT` class, which itself represents items of the "SoftcopyVOILUTSequence". Again, this actually represents a transformation, not a LUT.

With these purposes in mind, I looked for the common ground and called it `VOILUT`. I did not consider that this causes confusion with items of the `VOILUTSequence`, which appears under items of both "FrameVOILUTSequence" and "SoftcopyVOILUTSequence" and contains simply LUTs. So `VOILUT` is probably a bad name for this reason. One option is `VOITransformation`, but that seems odd as it would go into sequences called `...LUTSequence`.  The question in that case becomes what to do with it's base class `SoftcopyVOILUT`, should that become `SoftcopyVOITransformation`, despite being placed as items within "SoftcopyVOILUTSequence". Another option is `FrameVOILUT`, since this is the concrete item that it actually represents, then keep its subclass as it.

I do not see a purpose to creating a `ModalityTransformation` class. It does not currently exist and as far as I can tell this collection of elements does not exist anywhere within the standard except as a *module* called "Modality LUT" that places constraints saying that you must have either the rescale parameters or at LUT to represent a modality transform. It does not exist as an item within a sequence anywhere, as far as I am aware? I suppose we could implement it and have users create it and pass it to constructors, and then copy over the various attributes to the root level of the SOPClass dataset, like we have done for `PaletteColorLookupTable`, but I don't really see a strong reason why this would be better than how things are implemented currently. In my opinion it might make some sense to rename our private utility functions like `_add_modality_lut_attributes` -> `_add_modality_transformation_attributes`, but here we are definitely talking about the "Modality LUT" module, so maybe it's clearer as it?

I really don't see any reason to create a `VOILUT` class that is just a subclass of `LUT` without any changes. What is wrong with just passing a `LUT` where this is needed? I think the confusion here is that the existing `VOILUT` class does not do what you think.

Reply:
So in summary, I think we should either:
1. Rename `VOILUT` -> `FrameVOILUT`, or
2. Rename `VOILUT` -> `VOITransformation` AND `SoftcopyVOILUT` -> `SoftcopyVOITransformation` 

and leave it at that

Reply:
I think the major confusion comes from the fact that the same names are used for modules (collections of attributes, e.g., [Modality LUT Module Attributes](https://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.11.html#table_C.11-1)), macros (collections of attributes, e.g., [Modality LUT Macro Attributes](https://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.11.html#table_C.11-1b)), and individual attributes (e.g., [Modality LUT Sequence](https://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.11.html#para_8ed414d3-e3c3-449b-8407-6384126d0d32)).

It's even worse with VOI LUT, because there is a nesting of attributes with very similar names:
- [Softcopy VOI LUT Module Attributes](https://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.11.8.html#table_C.11.8-1)
- [Softcopy VOI LUT Sequence](https://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.11.8.html#para_f5f8feb7-aa86-4a93-aa6f-c9ec483d3b68)
- [VOI LUT Macro Attributes](https://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.11.2.html#table_C.11-2b)
- [VOI LUT Sequence](https://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.11.2.html#para_6cf459c0-7173-4ade-a142-9a28ee76ced9)

Now what does `VOILUT` implement? Currently, it implements the VOI LUT Macro Attributes. These are included in Softcopy VOI LUT Sequence, but should `SoftcopyVOILUT` therefore inherit from `VOILUT`?

So far, we have implemented sequence items by Python classes, but I am not aware that we implemented macros (we definitely didn't implement modules).

It's confusing as hell that `VOILUT` does not inherit from `LUT`, but `ModalityLUT` does, and that `VOILUT` contains the `WindowCenter` and `WindowWidth` attributes, but `ModalityLUT` does not contain `RescaleSlope` and `RescaleIntercept`. I think to get some sanity into all of this, we'll have to break with the standard.

The Modality LUT and VOI LUT information entities (yes, they also have the same name!) are part of the [Pixel Transformation Sequence](https://dicom.nema.org/medical/dicom/current/output/chtml/part04/sect_N.2.html). Therefore, I thought we could implement the **macros** by `ModalityLUTTransformation` and `VOILUTTransformation` Python classes.

Reply:
The standard even uses the terms "Modality LUT transformation" and "VOI LUT transformation" (see sections [N.2.1.1 Modality LUT](https://dicom.nema.org/medical/dicom/current/output/chtml/part04/sect_N.2.html#sect_N.2.1.1) and [N.2.1.3 VOI LUT](https://dicom.nema.org/medical/dicom/current/output/chtml/part04/sect_N.2.html#sect_N.2.1.3), respectively).

Reply:
Ok. Here is the implementation of the above proposal: 4d2c34b39183ee421246198e7ddc7f0511cbdc65

There are now the following classes for the different LUTs and the corresponding pixel transformations:
- `ModalityLUT` (inherits from `LUT`)
- `ModalityLUTTransformation`
- `VOILUT` (inherits from `LUT`)
- `VOILUTTransformation`
- `PresentationLUT` (inherits from `LUT`)
- `PresentationLUTTransformation`
- `ColorPaletteLUT` and `SegmentedColorPaletteLUT` (do **not** inherit from `LUT` because they require different attributes, but they have the same highdicom interface, i.e., same properties)
- `ColorPaletteLUTTransformation`

In addition, there is a `SoftcopyVOILUTTransformation` (inherits from `VOILUTTransformation`).